Mobile/starter_project_dark_knights/lib/features/signup/presentation/widgets/signup_form.dart

```diff
@@ -0,0 +1,109 @@
+import 'package:flutter/material.dart';
+import '../../../../core/utils/colors.dart';
+import 'navigation.dart';
+import 'buttons.dart';
+
+class SignUpForm extends StatefulWidget {
+  const SignUpForm({super.key});
+
+  @override
+  _SignUpFormState createState() => _SignUpFormState();
+}
+
+class _SignUpFormState extends State<SignUpForm> {
+  bool _showPassword = true; // To toggle password visibility
+
+  @override
+  Widget build(BuildContext context) {
+    final double screenHeight = MediaQuery.of(context).size.height;
+//MediaQuery to get the screen height and adjust the sizes of UI elements accordingly.
+    return Column(
+      children: [
+        Column(
+          children: [
+            const Align(
+              alignment: Alignment.centerLeft,
+              child: Text(
+                'Welcome',
+                style: TextStyle(
+                    fontWeight: FontWeight.w700,
+                    fontSize: 24,
+                    fontFamily: "Urbanist"),
+              ),
+            ),
+            SizedBox(height: screenHeight * 0.02),
+            const Align(
+              alignment: Alignment.centerLeft,
+              child: Text('Provide credentials to sign up',
+                  style: TextStyle(
+                      fontWeight: FontWeight.w800,
+                      fontFamily: "Poppins",
+                      color: secondaryTextColor)),
+            ),
+          ],
+        ),
+        SizedBox(height: screenHeight * 0.04),
+        //Username text field
+        const TextField(
```

Comment:
Make the username form field and the password form field the same component and pass in parameters. All of the text form fields in the screen can be one widget.

Reply:
Done